### DISCUSSION TITLE
problem to connect HM-CFG-USB2

### DISCUSSION THREAD

**Discussion by andrejsvet** [devRoemer/hassio-homegear-generic#5] [General]
I am trying to set up HM-cfg-usb2 adapter in home assistant (HA) in order to pair and control homematic devices (I would like to migrate from FHEM) and the only way is to use Homegear. I installed Virtualbox on Ubuntu and on VM I sucessfully installed HA ( core-2021.5.5), The next step was to install your HA addon Homegear and follow your instructions here https://github.com/devRoemer/hassio-homegear-generic/wiki. In the homematicbidcos.conf I also added the following configuration
[HM-CFG-USB]
id = hmland
# Uncomment this if you want this HM-CFG-USB to be your default communication module.
default = true
deviceType = hmcfglan
# IP address running the hmland service
host = 127.0.0.1
# Port number of the hmland service
port = 1234
responseDelay = 60
I also installed on Ubuntu the application, which emulates the HomeMatic LAN configuration adapter-protocol and created a service (https://git.zerfleddert.de/cgi-bin/gitweb.cgi/hmcfgusb). Unfortunately after I started the addon Homegear in HA the my USB gateway was not connected. I also checked in HomegearAdmin WEBUI and also the gateway was found but not connected. Can you please help on this matter? I am attachng part of  log from Homegear addon.

[homegear.log](https://github.com/devRoemer/hassio-homegear-generic/files/6586962/homegear.log)

**Comment by devRoemer** [comment]
Ah, i also worked with fhem before and now i am quite happy that i switched to ha. My setup is basing on a nanoCul adapter so i am not sure how  much i can help with the HM-CFG-USB2.

After taking a look at the log file i noticed that there are no errors. Only the message "Connecting to host 127.0.0.1 on port 1234..." appears quite often. From my experience i wold have expected a connection refused or a timeout error indicating that there are connection issues as you describe but maybe homegear just does not log in a way i am expecting. You could verify that if you change the host to a non existing ip on purpose => If there are no other entries it is how homegear logs connection issues.

If the lan emulator is running on your ubuntu vm i would not expect host 127.0.0.1 (localhost) in your homegear configuration. The homegear addon is running in a docker container where this ip means port 1234 of the container. What you instead need is the ip or hostname of the ubuntu vm with the emulator.

For troubleshooting you can also connect to the container with the following command:
docker ps -a
docker exec -it <id-of-the-container> /bin/sh

And try to reach services e.g. by executing the following:
curl 127.0.0.1:1234

(ping or telnet would be better but neither this nor that is available inside the container)

**Reply by andrejsvet** [reply]
Thank you Stefan for your quick answer. According to your proposal I changed in the  homematicbidcos.conf the host to the IP of the computer on which the ubuntu is running and gateway HM-CFG-USB2 was recognised and I was able to pair homematic devices. Thanks again!
